test/libsolidity/SolidityABIJSON.cpp

```diff
@@ -46,7 +46,7 @@ class JSONInterfaceChecker
 
 		Json::Value generatedInterface = m_compilerStack.contractABI("");
 		Json::Value expectedInterface;
-		m_reader.parse(_expectedInterfaceString, expectedInterface);
+		BOOST_REQUIRE(m_reader.parse(_expectedInterfaceString, expectedInterface));
```

Comment:
If a test throws an exception, it fails automatically. The `ETH_TEST_REQUIRE_NO_THROW` was only introduced to improve the error message, but it has some other nasty drawbacks.